src/Uno.UWP/ApplicationModel/Background/BackgroundTaskBuilder.Android.cs

```diff
@@ -0,0 +1,244 @@
+								break;
+							case SystemConditionType.InternetAvailable:
+								jobBuilder.SetRequiredNetworkType(Android.App.Job.NetworkType.Any);
+								break;
+							default:
+								throw new ArgumentException("BackgroundTaskBuilder.Register(): unimplemented Condition for TimeTrigger");
+						}
+					}
+
+					var jobScheduler =
+						(Android.App.Job.JobScheduler)Android.App.Application.Context.GetSystemService(
+							Android.Content.Context.JobSchedulerService);
+
+					var scheduleResult = jobScheduler.Schedule(jobBuilder.Build());
+
+					if (scheduleResult == Android.App.Job.JobScheduler.ResultFailure)
+					{
+						throw new ArgumentException("BackgroundTaskBuilder.Register(): Android JobScheduler returned failure");
+					}
+
+					#endregion
+					break;
+				case 2: // SystemTrigger
+					#region "SystemTrigger"
+					if (_broadcastReceiver is null)
+					{
+						throw new ArgumentException("BackgroundTaskBuilder.Register(): to register SystemTrigger, _broadcastReceiver has to be set - use UnoInitTriggers() method (probably you are using App.xaml.cs from older Uno version)");
+					}
+
+					// check subtype
+					switch ((_trigger as SystemTrigger).TriggerType)
+					{ // supported types
+						case SystemTriggerType.ServicingComplete:
+						case SystemTriggerType.SmsReceived:
+						case SystemTriggerType.TimeZoneChange:
+						case SystemTriggerType.UserAway:
+						case SystemTriggerType.UserPresent:
+							break;
+						default:    // all other types: unsupported
+							throw new NotImplementedException("Unimplemented type of SystemTrigger");
+					}
+
+					// maybe throw if any of _conditions is unimplemented in this type of trigger
+					foreach (SystemCondition condition in _conditions)
+					{
+						switch (condition.ConditionType)
+						{
+							default:
+								throw new ArgumentException("unimplemented Condition for SystemTrigger");
+						}
+					}
+
+					if (!_receiverRegistered)
+					{	// none of them is STICKY
+						var filter = new Android.Content.IntentFilter();
+						filter.AddAction(Android.Provider.Telephony.Sms.Intents.SmsReceivedAction);
+						filter.AddAction(Android.Content.Intent.ActionTimezoneChanged);
+						filter.AddAction(Android.Content.Intent.ActionUserPresent);
+						filter.AddAction(Android.Content.Intent.ActionMyPackageReplaced);
+						filter.AddAction(Android.Content.Intent.ActionScreenOff);
+
+						Android.App.Application.Context.RegisterReceiver(_broadcastReceiver, filter);
+						_receiverRegistered = true;
+					}
+
+					#endregion
+					break;
+
+				default:
+					throw new ArgumentException("BackgroundTaskBuilder.Register(): cannot register this trigger (unimplemented)");
+			}
+
+			// after all checking, we can store trigger data
+
+			// Doc says: "The chance that the value of the new Guid will be all zeros or equal to any other Guid is very low"
+			// so it is possible, and test for Empty costs us nothing
+			do
+			{
+				registration.TaskId = Guid.NewGuid();

```

Comment:
This is not needed: https://docs.microsoft.com/en-us/dotnet/api/system.guid.newguid?redirectedfrom=MSDN&view=netcore-3.1#remarks

Reply:
Yes, current version of doc says "The returned Guid is guaranteed to not equal Guid.Empty.". When I was creating this code, it was "The chance that the value of the new Guid will be all zeros or equal to any other Guid is very low." (see also https://softwareengineering.stackexchange.com/questions/274473/is-it-worth-even-checking-to-see-if-guid-newguid-is-guid-empty ).

Reply:
Maybe that's what it used to be, but it's not needed anymore. You can remove this.

Reply:
Are you sure that Mono used by Uno, in all versions, would not return Guid.Empty? This loop is very simple, so cost of using it is negligible.